### DISCUSSION TITLE
Data Start_values for maps

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#1886] [Q&A]
Is it possible to set data_start_values for maps?

**Comment by christoph-LSN** [comment]
Hi @sarah-beasley 

Maps support disaggregations as you can see here: https://christoph-lsn.github.io/MT_Site/1-2-4/

![image](https://user-images.githubusercontent.com/94443931/204976311-2da9d41f-10a6-4afe-be76-3e598326a51d.png)

Opening the map "Irak" is selected, but I want "Polen" is selected first opening the map

**Reply by sarah-beasley** [reply]
Thanks, I have checked with Brock and he has confirmed that while we do now have map disaggregations enabled, (as you are making great use of!) we do not have data_start_values for maps yet. This is something we could look into though for development of Open SDG.

**Reply by MoBosse** [reply]
If I may add something to the topic: We have [a case on our platform](https://dns-indikatoren.de/en/8-5-ab/) where we have a series with two disaggregations (20-64 years and 60-64 years). For 20-64 years we have data up to 2021, for 60-64 years the series currently ends in 2019, but the time slider runs until 2021 no matter which disaggregation is chosen. Since the map initially shows the 60-64 years disaggregation and the most recent year, the user sees a blank map when the map is opened.

When integrating start values for maps, maybe it is possible to add a rule that makes the maps only show those years where there are values for the chosen series/unit/disaggregation.

**Reply by sarah-beasley** [reply]
Thanks for letting us know about this - I have raised both problems as issues on github to be fixed - #1889 and #1890

**Comment by brockfanning** [comment]
@MoBosse @christoph-LSN When it comes to being able to specify start values for the map, which of these solutions do you think is better:
1. The existing "data_start_values" setting that already affects charts/tables is improved so that it also affects maps, or...
2. A new "data_start_values_maps" setting is added that only affects maps, and not charts/tables

One benefit of the first option is that existing configuration will automatically apply to the maps.
On the other hand, a benefit of the second option is that it would be possible to have different behavior for maps specifically.

I suppose it is possible that both could be added - in which case we might start with the first option and then add the second option afterwards. But, in your specific cases, do you think it would be problematic if the existiing data_start_values started affecting maps?

**Reply by MoBosse** [reply]
@brockfanning Hard to say. I can't think of a case where the defined "data_start_values" shouldn't be good for the map as well. 
But if possible, why not combine the two options so that "data_start_values" are used when "data_start_values_maps" are not defined?
